indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx

```diff
@@ -370,6 +398,27 @@ export default function SingleChoiceInput({
   placesUsed,
 }) {
   const existingValue = useSelector(state => getFieldValue(state, fieldId)) || {};
+
+  const optionSet = useMemo(() => new Set(choices.map(choice => choice.id)), [choices]);
+
+  function validate(value) {
+    const noValue = !value || !Object.keys(value).length;
+    if (isRequired && noValue) {
+      return Translate.string('This field is required');
+    }
+
+    if (noValue) {
+      // When there is no value but the field is not required, it's a pass
+      return;
+    }
+
+    for (const key in value) {
+      if (!optionSet.has(key)) {
+        return Translate.string('Please select one of the provided choices');
```

Comment:
Can this still happen? I think it might be a leftover from when it was a ComboBox